### PR TITLE
Add "kind" field to app service.

### PR DESCRIPTION
**Pull request by JunyiYi**
Enable "kind" field in app service.
Now you are able to create FunctionsApp (but you still need to care about other configurations).

### CODE REVIEW

File: azurerm/resource_arm_app_service.go [hunk 1]
```diff
@@ -33,6 +33,16 @@ func resourceArmAppService() *schema.Resource {
 
 			"location": locationSchema(),
 
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"functionapp",
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service.go]: we should add all the other values in here too (from memory there's 4 values - `functionapp`, `mobileapi` and `app` and something else?) - there should be constants for these in the Azure SDK.

File: azurerm/resource_arm_app_service.go [hunk 2]
```diff
@@ -33,6 +33,16 @@ func resourceArmAppService() *schema.Resource {
 
 			"location": locationSchema(),
 
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service.go]: we should add a Default value for this for existing users, e.g. "app" (which appears to be the default for a Web App/Website/App Service)

Comment (JunyiYi) [reply azurerm/resource_arm_app_service.go]: Sure. And where did your find those available values? My knowledge about "functionapp" is provided by the Azure Functions team.

File: azurerm/resource_arm_app_service.go (hunk 1 above)
Comment (JunyiYi) [reply azurerm/resource_arm_app_service.go]: Will do if I could confirm the available values.

File: azurerm/resource_arm_app_service.go (hunk 2 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_app_service.go]: from memory creating an App Service / Function App / Mobile API in the portal and either seeing it in the output of resources.azure.com / API responses / the export template function - although they may also be in the Go SDK

**Comment by JunyiYi** [comment]
Thanks @tombuildsstuff ,

This is the way recommended by Azure Functions team: "Set kind to functionapp, and remember to configure the required storage as well as connection string app settings." And there are no dedicated SDK calls or API's for Azure Functions right now.

I agree that users should not care about the detailed difference between different "app"s, so based on this, we would also like to add a new module maybe called azure functions, and users should use that module to provision an azure functions app. And the provider here will just provide the essential building blocks.

Regards.

**Pull request closed without merging**